src/utils.jl

```diff
@@ -77,6 +77,55 @@ function sideof(p::Point{2,T}, c::Chain{2,T}) where {T}
   ifelse(isapprox(w, zero(T), atol=atol(T)), :OUTSIDE, :INSIDE)
 end
 
+"""
+    sideof(point, mesh)
+
+Determines whether a `point` is inside, outside or on the surface of a `mesh`.
+Possible results are `:INSIDE`, `:OUTSIDE`, or `:ON`.
+
+### Notes
+
+Uses a ray-casting algorithm.
+"""
+function sideof(point::Point{3,T}, mesh::Mesh{3,T}) where {T}
+  if !(eltype(mesh) <: Triangle)
+    throw(ArgumentError(
+          "This function only works for surface meshes with triangles as elements."))
+  end
+
+  z = last.(coordinates.(extrema(mesh)))
+  r = Ray(point, Vec(zero(T), zero(T), 2 * (z[2] - z[1])))
+
+  intersects = false
+  edgecrosses = 0
+  ps = Point{3,T}[]
+  for t in mesh
+    result = intersection(r, t) do I
+      if type(I) == CrossingRayTriangle
+        intersects = !intersects
+      elseif type(I) ∈ (EdgeTouchingRayTriangle,
+                        CornerTouchingRayTriangle,
+                        TouchingRayTriangle)
+        return :ON
+      elseif type(I) == EdgeCrossingRayTriangle
+        edgecrosses += 1
+      elseif type(I) == CornerCrossingRayTriangle
+        p = get(I)
+        if !any(==(p), ps)
+            push!(ps, p)
+            intersects = !intersects
+        end
+      end
+      return nothing
+    end
+    !isnothing(result) && return :ON
+  end
+
+  # check how many edges we crossed
+  isodd(round(Int, edgecrosses / 2)) && (intersects = !intersects)
```

Comment:
Do you edgecrosses \div 2 here by any chance? Or you really need to round for correctness?

Reply:
Yes I forgot about \div.